### PR TITLE
completing task 11

### PR DESCRIPTION
**Pull request by A-E-Harding**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -365,3 +367,125 @@ describe("PATCH /api/articles/:article_id", () => {
...
+})
+
+describe("GET /api/articles?order=:order", () => {
+  test('200: sets order to asc or desc', () => {
+    return request(app)
+      .get("/api/articles?order=asc")
+    .expect(200)
+      .then((response) => {
+        let articles = response.body
+        expect(articles).toBeSortedBy('created_at', { ascending: true })
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: I don't think ascending is an option, I think this only works because descending defaults to false

File: app.js [hunk 2]
```diff
@@ -26,6 +26,12 @@ app.patch("/api/articles/:article_id", updateVotes);
 
 app.get('/api/users', getAllUsers)
 
+app.get("/api/articles?topic=:topic", getAllArticles)
+
+app.get("/api/articles?sort_by=:property", getAllArticles)
+
+app.get("/api/articles?order=:order", getAllArticles)
+
```
Comment (Hard-Luck) [inline app.js]: queries don't affect the path matching so /api/articles will match with any number of queries. This means you only need one match for app.get("/api/articles")

File: models/models.js [hunk 3]
```diff
@@ -19,14 +19,40 @@ exports.readArticles = (article_id) => {
...
-    .query(
-      "SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM ARTICLES LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC"
-    )
+exports.readAllArticles = (topic, sort_by = 'created_at', order="desc") => {
+  const acceptedTopics = ["mitch", "cats", "paper", undefined];
+  const queryValues = [];
+  const acceptedProperties = ['title', 'topic', 'author', 'body', 'created_at', 'votes'];
+  const acceptedOrders = ['asc', 'desc']
+
+  if (!acceptedTopics.includes(topic)) {
```
Comment (Hard-Luck) [inline models/models.js]: You don't need to bad request if the topic is not found, it doesn't need to be a rigid as the others as this is not a column name and can be easily escaped using $1 syntax, so there is no risk of SQL injection therefore no need to have a list of acceptable topics.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ const testData = require("../db/data/test-data/index");
 const seed = require("../db/seeds/seed");
 const connection = require("../db/connection");
 const endpoints = require("../endpoints.json");
+const { response } = require("../app.js");
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: Is this a mistake?

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Main change required is the multiple handlers for the same endpoint

**Review by Hard-Luck — APPROVED** [review]
I would add one test to ensure that you get an empty array when topics isn't matched, then merge!

Great work
